### PR TITLE
Styled List Items

### PR DESCRIPTION
**Pull request by guilean**
**Checklist**
- [X] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)
https://trello.com/c/mMYjbJyj

### CODE REVIEW

**Comment by tkanzakic** [comment]
merge conflicts

File: packages/styled-components/src/components/primitives/List/Items/ButtonItem/ButtonItem.test.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { ButtonItem } from '.';
+
+describe('ButtonItem', () => {
+  const testId = 'buttonItem';
+  it('should render correctly with children', () => {
+    render(
+      <ButtonItem>
+        <div>foo</div>
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/List/Items/ButtonItem/ButtonItem.test.tsx:10]: in other components we only used the text and asserted using getByText

File: packages/styled-components/src/components/primitives/List/Items/ButtonItem/ButtonItem.test.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    );
+    const buttonItem = screen.getByTestId(testId);
+    expect(buttonItem.children).toHaveLength(1);
+  });
+  it('should render class', () => {
+    render(<ButtonItem className="foo" />);
+    const buttonItem = screen.getByTestId(testId);
+    expect(buttonItem).toHaveClass('foo');
+  });
+  it('should call onClick on click', () => {
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/List/Items/ButtonItem/ButtonItem.test.tsx:21]:
```suggestion
  it('should call onClick on clicked', () => {
```

File: packages/styled-components/src/components/primitives/List/Items/LinkItem/LinkItem.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { text, select } from '@storybook/addon-knobs';
+
+export default {
+  title: 'Primitives/List/Items/LinkItem',
+  component: LinkItem,
+};
+
+export const Icons = () => (
+  <LinkItem icon={<Favorite type={Icon.Types.Filled} />} title="Title" />
+);
```
Comment (tkanzakic) [inline packages/styled-components/src/components/primitives/List/Items/LinkItem/LinkItem.stories.tsx:14]: what about the other cases?

**Review by tkanzakic — CHANGES_REQUESTED** [review]

File: packages/styled-components/src/components/primitives/List/Items/LinkItem/LinkItem.stories.tsx (hunk 3 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/List/Items/LinkItem/LinkItem.stories.tsx]: Added

File: packages/styled-components/src/components/primitives/List/Items/ButtonItem/ButtonItem.test.tsx (hunk 1 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/List/Items/ButtonItem/ButtonItem.test.tsx]: Done

File: packages/styled-components/src/components/primitives/List/Items/ButtonItem/ButtonItem.test.tsx (hunk 2 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/List/Items/ButtonItem/ButtonItem.test.tsx]: Done

File: packages/styled-components/src/components/primitives/List/Items/LinkItem/LinkItem.stories.tsx (hunk 3 above)
Comment (guilean) [reply packages/styled-components/src/components/primitives/List/Items/LinkItem/LinkItem.stories.tsx]: Done

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**